LearningMissionSimulation/PlaygroundMher.cs

```diff
@@ -197,58 +264,106 @@ List<Module> GetModuleList()
-                    classroomList.Add(classroom);
+                    _classroomList.Add(classroom);
                     i++;
+
+                    ReportItem(itemName: classroom.ToString(), actionName: action, itemIndex: i);
                 }
-                ReportSummary("Created", "Classrooms", i);
-                ReportFooter("Create Classroom");
+
+                ReportSummary(actionName: action, itemCount: i);
+                ReportFooter(actionName: action);
             }
         }
 
         public void AssignInstructorsToClassrooms()
         {
-            if(classroomList.Count == 0)
+            string action = "Assign Instructors To Classrooms";
+            ReportHeader(actionName: action);
+
+            if (_classroomList.Count == 0)
             {
-                ReportError("Instructor", "Classroom");
+                ReportError(missingResource: "Instructor", failedAction: action);
             }
             else
             {
-                foreach (var classroom in classroomList)
+                foreach (var classroom in _classroomList)
                 {
-                    if (moduleInstructorDictionary.ContainsKey(classroom.Module.Id))
+                    if (_moduleInstructorDictionary.ContainsKey(classroom.Module.Id))
                     {
                         List<Instructor> instructorList;
-                        moduleInstructorDictionary.TryGetValue(classroom.Module.Id, out instructorList);
+                        _moduleInstructorDictionary.TryGetValue(classroom.Module.Id, out instructorList);
                         classroom.Head = instructorList[AttributeGenerator.random.Next(0, instructorList.Count)];
                     }
                     else
                     {
-                        ReportError("Instructor", "Classroom");
+                        ReportError(missingResource: "Instructor", failedAction: action);
                     }
                 }
             }
+
+            ReportFooter(actionName: action);
         }
 
         public void RegisterStudentsForClasses()
         {
-           
+            string action = "Register Students For Classes";
+            ReportHeader(actionName: action);
+
+            if (_classroomList.Count == 0)
+            {
+                ReportError(missingResource: "Instructor", failedAction: action);
+            }
+            else
+            {
+                foreach (var classroom in _classroomList)
+                {
+                    if(classroom.ItemList.Count < classroom.MaximumCapacity)
+                    {
+                        UpdateStudentList(classroom);
+                    } 
+                   
+                }
+            }
+
+            ReportFooter(actionName: action);
+        }
+
+
+        void UpdateStudentList(Classroom classroom)
+        {
+            int itemListCount = AttributeGenerator.random.Next(0, classroom.MaximumCapacity - classroom.ItemList.Count);
```

Comment:
the upper limit should be classroom.MaximumCapacity - classroom.ItemList.Count + 1